test/utils/custom_tsv_parser.py

```diff
@@ -0,0 +1,50 @@
+import csv
+
+
+class ParseTSVFile:
+    def __init__(self, file_path):
+        self.file_path = file_path
+
+    def read_file(self):
+        with open(self.file_path, "r", newline="", encoding="utf-8") as file:
+            reader = csv.DictReader(file, delimiter="\t")
+            return reader
+
+    def get_columns(self):
+        with open(self.file_path, "r", newline="", encoding="utf-8") as file:
+            reader = csv.DictReader(file, delimiter="\t")
+            return reader.fieldnames
+
+    def get_row_count(self):
+        with open(self.file_path, "r", newline="", encoding="utf-8") as file:
+            reader = csv.DictReader(file, delimiter="\t")
+            return len(list(reader))
+
+    def get_column_count(self):
+        with open(self.file_path, "r", newline="", encoding="utf-8") as file:
+            reader = csv.DictReader(file, delimiter="\t")
+            return len(reader.fieldnames)
+
+    def get_rows(self):
+        with open(self.file_path, "r", newline="", encoding="utf-8") as file:
+            reader = csv.DictReader(file, delimiter="\t")
+            data = []
+            for row in reader:
+                data.append(row)
+            return data
+
+    def get_column_data(self, column_name):
```

Comment:
These seem unnecessary functions; it would be good to not add any functions that we don't need. This becomes unmanageable down the line and then we have functions that don't have tests also which is going to be pretty heavy for anyone to manage these

Reply:
I beg to differ though. These functions are decoupled and maintained inside utils. In this way we have many other objects inherit the same functionality with less repeated code.

Reply:
Hmm, are we sure that this function is used elsewhere? Otherwise, leaving around unused and untested functions is a relatively bad idea.